imu_sensor_broadcaster/include/imu_sensor_broadcaster/imu_sensor_broadcaster.hpp

```diff
@@ -49,18 +51,25 @@ class IMUSensorBroadcaster : public controller_interface::ControllerInterface
   controller_interface::CallbackReturn on_deactivate(
     const rclcpp_lifecycle::State & previous_state) override;
 
-  controller_interface::return_type update(
+  controller_interface::return_type update_and_write_commands(
     const rclcpp::Time & time, const rclcpp::Duration & period) override;
 
+  controller_interface::return_type update_reference_from_subscribers(
+    const rclcpp::Time & time, const rclcpp::Duration & period) override;
+
+  std::vector<hardware_interface::StateInterface> on_export_state_interfaces() override;
+
 protected:
   std::shared_ptr<ParamListener> param_listener_;
   Params params_;
+  Eigen::Quaternion<double> r_;
```

Comment:
```suggestion
  Eigen::Quaterniond r_;
```

There is already an internal typdef mapping that to d
https://eigen.tuxfamily.org/dox-devel/classEigen_1_1Quaternion.html

Not a big deal, if this is not applied.